crates/block-producer/src/block_builder/mod.rs

```diff
@@ -223,20 +252,93 @@ impl BlockBuilder {
         &self,
         mempool: &SharedMempool,
         built_block: ProvenBlock,
-    ) -> Result<(), BuildBlockError> {
+    ) -> Result<StateDelta, BuildBlockError> {
         self.store
             .apply_block(&built_block)
             .await
             .map_err(BuildBlockError::StoreApplyBlockFailed)?;
 
-        mempool.lock().await.commit_block();
-
-        Ok(())
+        let reverted_transactions = mempool.lock().await.commit_block();
+        let committed_transactions = built_block
+            .transactions()
+            .as_slice()
+            .iter()
+            .map(TransactionHeader::id)
+            .collect();
+        let committed_notes = built_block
+            .output_notes()
+            .filter_map(|(_idx, note)| {
+                // Only retain network notes.
+                if let OutputNote::Full(note) = note {
+                    if NoteExecutionMode::Network != note.metadata().tag().execution_mode() {
+                        return None;
+                    }
+                    // TODO: ensure it is a network note.
+                    CommittedNote {
+                        block_num: built_block.header().block_num(),
+                        note_index: todo!("Do I really have to iterate over the batches?"),
+                        note_id: note.id().inner(),
+                        metadata: *note.metadata(),
+                        details: NoteDetails::from(note).to_bytes().into(),
+                        merkle_path: todo!("How do this?"),
```

Comment:
We can probably simplify this for now (and maybe forever) and not include Merkle path with notes. This way, NTX builder would always create transactions with unauthenticated notes, and then these would be authenticated during batch building.

This also implies that we probably don't need to include most fields here and can just send a simple `Note` (which would consist of `NoteDetails` and `NoteMetadata`).